### ISSUE TITLE
Card-mod cards causes all dashboard cards to flicker after the HA update when Entity-filter in the same dashboard." 

### ISSUE THREAD

**Issue by RobusX** [thomasloven/lovelace-card-mod#407]
My Home Assistant version: 2024.9.3

My lovelace configuration method (GUI or yaml): Yaml

After updating from 2024.6.4 to 2024.9.3, all dashboard cards with modified colors or card-mod hidden buttons are flickering (colors and hidden buttons blink) when an entity-filter card with color-changing mode is also present on the dashboard. The issue cannot be reproduced in safe mode, as no cards are displayed.

Flickering appeared with Firefox, Google Chrome and also within latest HA Android Companion app.
Slider button on the right browser side also blinks.

If the dashboard displays cards in the bottom part, browser due to flickering issue moves to the upper dashboard side, where the right slider also blinks.
As was discovered, issue disappears when entity-filter card is removed from the dashboard or dashboard in edit mode.
The problem is very annoying; please resolve it. The uses of the card-mod card are endless, and it would be a pitty to abandon it.

"Please review the detailed issue description and dashboard card codes (link below), where it was stated from the frontend developer that the problem is caused by card-mod not being updated to follow the latest HA frontend releases."

https://github.com/home-assistant/frontend/issues/22084

**By putting an X in the boxes ([]) below, I indicate that I:**

- X Understand that this is a channel for reporting bugs, not a support forum (https://community.home-assistant.io/).
- X Have made sure I am using the latest version of the plugin.
- X Have followed the troubleshooting steps of the "Common Problems" section of https://github.com/thomasloven/hass-config/wiki/Lovelace-Plugins.
- X Understand that leaving one or more boxes unticked or failure to follow the template above may increase the time required to handle my bug-report, or cause it to be closed without further action.

**Comment by ildar170975** [comment]
Please post a SHORT minimal code to reproduce the issue.

**Comment by RobusX** [comment]
Yes, no problem.

Firstly, there are two automations which generate blinking lights for aqara lights (2 fields - columns in the entitiy-filter card -> please accommodate it to your entities and remove alarm/sun conditions, only blinking is needed:

alias: Light Night Blinking RED when Armed
description: Light Night Blinking RED when Armed
trigger:
  - platform: state
    entity_id:
      - switch.alarm_plug
    to: "on"
    id: alarm
  - platform: numeric_state
    entity_id: sun.sun
    value_template: "{{ state_attr('sun.sun', 'elevation') }}"
    below: -3
    id: sunset
  - platform: numeric_state
    entity_id: sun.sun
    value_template: "{{ state_attr('sun.sun', 'elevation') }}"
    above: -3
    id: sunrise
  - platform: time_pattern
    minutes: /3
    id: alarm
condition:
  - condition: or
    conditions:
      - condition: and
        conditions:
          - condition: state
            entity_id: switch.alarm_plug
            state: "on"
          - condition: trigger
            id: sunset
          - condition: time
            after: "18:00:00"
            before: "07:00:00"
      - condition: and
        conditions:
          - condition: state
            entity_id: switch.alarm_plug
            state: "on"
          - condition: trigger
            id: sunrise
          - condition: time
            after: "06:30:00"
            before: "18:00:00"
      - condition: and
        conditions:
          - condition: state
            entity_id: switch.alarm_plug
            state: "on"
          - condition: trigger
            id: alarm
action:
  - choose:
      - conditions:
          - condition: trigger
            id: sunset
        sequence:
          - repeat:
              sequence:
                - target:
                    entity_id:
                      - light.aqara_hub_m1s_5fc8_lightbulb
                      - light.aqara_hub_m1s_64d3_lightbulb
                  data:
                    color_name: red
                    brightness_pct: 70
                  action: light.turn_on
                - delay: "00:00:00.6"
                - target:
                    entity_id:
                      - light.aqara_hub_m1s_5fc8_lightbulb
                      - light.aqara_hub_m1s_64d3_lightbulb
                  data:
                    brightness_pct: 3
                    kelvin: 3800
                  action: light.turn_on
                - delay: "00:00:05"
              until:
                - condition: state
                  entity_id: switch.alarm_plug
                  state: "off"
      - conditions:
          - condition: trigger
            id: sunrise
        sequence:
          - repeat:
              sequence:
                - target:
                    entity_id:
                      - light.aqara_hub_m1s_5fc8_lightbulb
                      - light.aqara_hub_m1s_64d3_lightbulb
                  data:
                    color_name: red
                    brightness_pct: 70
                  action: light.turn_on
                - delay: "00:00:00.6"
                - target:
                    entity_id:
                      - light.aqara_hub_m1s_5fc8_lightbulb
                      - light.aqara_hub_m1s_64d3_lightbulb
                  data:
                    brightness_pct: 3
                    kelvin: 3800
                  action: light.turn_on
                - delay: "00:00:05"
              until:
                - condition: state
                  entity_id: switch.alarm_plug
                  state: "off"
      - conditions:
          - condition: trigger
            id: alarm
        sequence:
          - repeat:
              sequence:
                - target:
                    entity_id:
                      - light.aqara_hub_m1s_64d3_lightbulb
                      - light.aqara_hub_m1s_5fc8_lightbulb
                  data:
                    color_name: red
                    brightness_pct: 70
                  action: light.turn_on
                - delay: "00:00:00.6"
                - target:
                    entity_id:
                      - light.aqara_hub_m1s_64d3_lightbulb
                      - light.aqara_hub_m1s_5fc8_lightbulb
                  data:
                    brightness_pct: 3
                    kelvin: 3800
                  action: light.turn_on
                - delay: "00:00:05"
              until:
                - condition: state
                  entity_id: switch.alarm_plug
                  state: "off"
mode: restart


And here is other automation needed which blinks aditionally 2 kuril bulbs in the entity-filter card (when those bulbs blink, there are four columns in the entity-filter card (2 for aqara+2x kuril), so entity-filter card is stretching from 2 to 4 fields:

alias: Light Kuril Blinking RED when Armed
description: Light Kuril Blinking RED when Armed
trigger:
  - platform: state
    entity_id:
      - switch.alarm_plug
    to: "on"
    id: day
  - platform: time_pattern
    minutes: /3
    id: day
condition:
  - condition: state
    entity_id: switch.alarm_plug
    state: "on"
action:
  - choose:
      - conditions:
          - condition: trigger
            id: day
        sequence:
          - repeat:
              sequence:
                - target:
                    entity_id:
                      - light.kurilbulb1
                      - light.kurilbulb2
                  data:
                    color_name: red
                    brightness_pct: 2
                  action: light.turn_on
                - delay: "00:00:00.1"
                - target:
                    entity_id:
                      - light.kurilbulb1
                      - light.kurilbulb2
                  data: {}
                  action: light.turn_off
                - delay: "00:00:00.1"
                - target:
                    entity_id:
                      - light.kurilbulb1
                      - light.kurilbulb2
                  data:
                    color_name: white
                    brightness: 2
                  action: light.turn_on
                - delay: "00:00:00.1"
                - target:
                    entity_id:
                      - light.kurilbulb1
                      - light.kurilbulb2
                  data: {}
                  action: light.turn_off
                - delay: "00:00:06"
              until:
                - condition: state
                  entity_id: switch.alarm_plug
                  state: "off"
          - data:
              brightness_pct: 20
              kelvin: 3500
            target:
              entity_id:
                - light.kurilbulb1
                - light.kurilbulb2
            action: light.turn_on
          - delay: "00:00:01"
          - data: {}
            target:
              entity_id:
                - light.kurilbulb1
                - light.kurilbulb2
            action: light.turn_off
mode: single

And here is entity-filter card for the dashboard which contains those four bulbs:

type: entity-filter
title: Lights/Sensors ON
show_header_toggle: false
entities:
  - entity: light.aqara_hub_m1s_5fc8_lightbulb
    name: AqSFlL
    tap_action:
      action: toggle
  - entity: light.aqara_hub_m1s_64d3_lightbulb
    name: AqStrL
    tap_action:
      action: toggle
  - entity: light.kurilbulb1
    name: Kur_1
    tap_action:
      action: toggle
  - entity: light.kurilbulb2
    name: Kur_2
    tap_action:
      action: toggle
state_filter:
  - 'on'
  - open
  - detected
  - operator: '>'
    value: 4
show_empty: false
sort:
  method: domain
card:
  type: glance
  title: ' '


And finally, here we have other (two for example) entities card in the same dashboard, which contains some car-mod code and are flickering (NOK):


1. Here on-off switch on the right near "Main Door" text, which is hidden by  card-mod, flickers:

type: entities
entities:
  - type: custom:template-entity-row
    entity: binary_sensor.contact_sensor
    name: Main Door State
    toggle: false
    secondary_info: last-updated
    secondary: >
      {% set state = states('binary_sensor.contact_sensor') %} {% if state ==
      'off' %}
        Closed
      {% elif state == 'on' %}
        Open
      {% else %}
        {{ state }}
      {% endif %} | Last: {{
      states.binary_sensor.contact_sensor.last_updated.astimezone().strftime('%m-%d
      %H:%M:%S') }}
    card_mod:
      style: |
        :host {
          --mdc-icon-size: 40px;    
          --paper-item-icon-color: {%
            if is_state('binary_sensor.contact_sensor', 'off') %}
            green;
          {%- elif is_state('binary_sensor.contact_sensor', 'on') or
              is_state('switch.shelly_main_door_12v_switch_0', 'on') %}
            red;
          {%- else %}
            grey;
          {%- endif %};
              }
  - type: custom:fold-entity-row
    head:
      entity: switch.shelly_main_door_12v_switch_0
      name: Main Door
      toggle: false
      state_color: true
      card_mod:
        style: |
          :host {
            --card-mod-icon-color: 
            {% if is_state(config.entity, 'on') %}
              orange;
            {% else %}
              grey;
            {% endif %};
          }
          .entities .state-card-toggle .state-card {
            display: none !important;
          }
          ha-entity-toggle {
            display: none !important;
          }
    entities:
      - type: custom:template-entity-row
        entity: switch.shelly_main_door_12v_switch_0
        name: Main Door 12V Switch
        tap_action:
          action: toggle
        toggle: true
        secondary_info: last-updated
        secondary: >
          Last: {{
          states.switch.shelly_main_door_12v_switch_0.last_updated.astimezone().strftime('%m-%d
          %H:%M:%S') }}
      - type: custom:template-entity-row
        entity: switch.2pm_main_door_switch_1
        name: Main Door 230V Supply
        tap_action:
          action: toggle
        toggle: true
        secondary_info: last-updated
        secondary: >
          Power: {{ states('sensor.2pm_main_door_switch_1_power') }} W |  Last:
          {{
    states.switch.shelly_main_door_12v_switch_0.last_updated.astimezone().strftime('%m-%d
          %H:%M:%S') }} 
title: ''
state_color: true
show_header_toggle: false

2. And in this entities card on the same dashboard all three doors, colored with green when closed (car-mod) are flickering:

type: entities
title: ' '
show_header_toggle: false
state_color: true
entities:
  - type: custom:template-entity-row
    entity: binary_sensor.contact_sensor
    name: Main Door
    icon: mdi:door
    style: |
      :host {
        --mdc-icon-size: 30px;
        --card-mod-icon-color: 
          {% if is_state(config.entity, 'on') %}
            red
          {% elif is_state(config.entity, 'unavailable') or is_state(config.entity, 'unknown') %}
            yellow
          {% else %}
            green
          {% endif %};
      }
    secondary: |
      {% if is_state(config.entity, 'on') %}
        Door Open
      {% elif is_state(config.entity, 'off') %}
        Door Closed
      {% elif is_state(config.entity, 'unavailable') %}
        Sensor Unavailable
      {% elif is_state(config.entity, 'unknown') %}
        Sensor State Unknown
      {% endif %}
  - type: custom:template-entity-row
    entity: binary_sensor.contact_sensor_3
    name: App Door
    icon: mdi:door
    style: |
      :host {
        --mdc-icon-size: 30px;
        --card-mod-icon-color: 
          {% if is_state(config.entity, 'on') %}
            red
          {% elif is_state(config.entity, 'unavailable') or is_state(config.entity, 'unknown') %}
            yellow
          {% else %}
            green
          {% endif %};
      }
    secondary: |
      {% if is_state(config.entity, 'on') %}
        Door Open
      {% elif is_state(config.entity, 'off') %}
        Door Closed
      {% elif is_state(config.entity, 'unavailable') %}
        Sensor Unavailable
      {% elif is_state(config.entity, 'unknown') %}
        Sensor State Unknown
      {% endif %}
  - type: custom:template-entity-row
    entity: binary_sensor.contact_sensor_2
    name: Cellar Door
    icon: mdi:door
    style: |
      :host {
        --mdc-icon-size: 30px;
        --card-mod-icon-color: 
          {% if is_state(config.entity, 'on') %}
            red
          {% elif is_state(config.entity, 'unavailable') or is_state(config.entity, 'unknown') %}
            yellow
          {% else %}
            green
          {% endif %};
      }
    secondary: |
      {% if is_state(config.entity, 'on') %}
        Door Open
      {% elif is_state(config.entity, 'off') %}
        Door Closed
      {% elif is_state(config.entity, 'unavailable') %}
        Sensor Unavailable
      {% elif is_state(config.entity, 'unknown') %}
        Sensor State Unknown
      {% endif %}   

I hope it is more clear now. Thanks for looking into it  !

**Comment by ildar170975** [comment]
Sorry, this is not a short MWE.

**Comment by ildar170975** [comment]
Also, the posted code is not formatted and cannot be analyzed.

**Comment by ildar170975** [comment]
Also, the posted code is not formatted and cannot be analyzed.

**Comment by RobusX** [comment]
I shorted code as much i could..but entities for blinking should be present in the dashboard and also in both automations. And also in the flickering cards.
I pasted yaml code for both automations, entity-filter card and both entities cards (flickering) and save it into notepad (.yaml file not supported to enclose). AS i see formatting is preserved, although txt format. I hope, this is ok.
Thank you for your effort.


[cards.txt](https://github.com/user-attachments/files/17194293/cards.txt)

**Comment by pickonedev** [comment]
Here you are :-D

I have a sample as well.
No matter what card I use, it will flicker when something is changed on the dashboard, for all the cards which uses card mod.

https://github.com/user-attachments/assets/d6b1b028-c80b-48c6-8e14-1d97a198f5cb

**Comment by RobusX** [comment]
Thanks @ pickonedev .
Can you confirm too that in dashboard editing mode it is no flickering at all ?
Thanks for the info.

**Comment by pickonedev** [comment]
That is right.

**Comment by RobusX** [comment]
Yes, today i also tried your case. I turned one switch to ON in the entities card and all cards that use card-mod code flickered at the same time. 
So problem appears not connected with the entity-filter card, if this is the same cause, as my other issue when entity-filter is present.

**Comment by pickonedev** [comment]
I converted all my dashboards to Sections and the issue partially dissapeared...
Now it is only happening to the entity-filter cards, in the past (when I was not using Sections), it would flicker all the cards where card_mod was being used

Btw ... does someone care about this issue?
I see no improvement or not even an official answer about this, for almost two months....

Anyway, even with an answer, I don't think this can be repaired, because this is the functionality way of the card_mod... After element was loaded, card_mod is applied, so... there is a gap of time till card_mod can be applied (depends of the performance of the device). For my old tablets, I can wait even 2-4 secunds... luckly that I change them and now the time is reduced to few ms... but still... it is very annoying to see that flicker... This is the same as loading a popup with elements modified by card_mod, time gap till everything appear properly...

**Comment by RobusX** [comment]
I am still experiencing the same issue with entity-filter, and yes, apparently nobody cares about it. Maybe most people no longer use entities cards. Partly, I replaced entity-filter with the custom:auto-entities card, and on those tabs, the situation is better.
In the meantime, I reported another issue related to the custom fold-entity-row card not opening its instances here:
[GitHub Issue](https://github.com/thomasloven/lovelace-fold-entity-row/issues/251#issuecomment-2472994871).
A similar situation applies—no one seems capable or willing to solve issues with "old" cards. It seems like we old-timers have to modify our dashboards to newer formats every year or two due to such issues (like how you transitioned your dash to "Sections.") While this may be a comprehensive update, it's likely not easy.
Did you use any converter tool to perform this transformation? Where did you find good instructions, for instance?
I am currently avoiding modifying my dashboards to use "Sections" due to the increased space they require (especially in the Android phone companion app, where I use the custom fold-entity-row to hide similar entities under a main one).
Another concern is how to display all secondary information (e.g., power, states, etc.) using "Sections." Currently, I use card-mod for this purpose.
Thanks in advance for any useful information!

**Comment by ildar170975** [comment]
> no one seems capable or willing to solve issues with "old" cards

Custom fold-entity-row, stock entity-filter - not "old cards".

As for custom cards which have issues:
1. [Particularly for f-e-r](https://github.com/thomasloven/lovelace-fold-entity-row/issues/251): this is about a particular Android browser (or webview, whatever) which may not be updated; similar things happen to old iOS devices which stop getting updates from Apple (not to mention that Safari is buggy & does not follow some conventions by default).
2. For some custom cards: their authors may be rather tired to follow changes in HA frontend & give up updating their cards (ofc some authors stop updating due to other personal reasons).

In my experience, most of issues with custom cards may be either "workarounded" or fixed by card-mod etc.

As for stock cards: entity-filter is not related to Sections. It just can be shown/not shown; can have height 100px or 200px. And these things are not according to "fancy sections way". That is why many experienced users do not use Sections - it is mainly a toy for beginners.

**Comment by RobusX** [comment]
@ildar170975
Thank you for the information regarding sections. It seems they might not be suitable for my use case, so I’ll stick with entities for as long as they work acceptable.

Regarding the Android issue with the fold entity row:
My phone always runs the latest version of Android. This issue appears inconsistently-sometimes it happens, and other times it does not. Therefore, I don’t believe it’s related to an outdated or buggy Android version. I suspect it’s related to recent Home Assistant changes affecting dashboards, including in the Companion App.

Regarding the entity filter card:
As I mentioned earlier, I can’t provide a more minimal working example than the one I already shared.
The issue occurs when two lights (named "kuril bulb1/2" in the automation) are blinking rapidly. In this case, the entity filter should expand and contract at the same rhythm as the blinking appeared (100ms delay above, many times interchangeable (one bulb or another or both bulbs displayed). This behavior used to work fine in previous versions of Home Assistant.

The "kuril blinking" automation is integral; without it, the lights wouldn’t blink. 
As a result of blinking, the entitiy filter card stretches and contracts . The entity filter also displays other bulbs, so it’s not entirely empty at the moment when the "kuril bulbs" are not shown. 
Other entity cards, which are flickering due to the enitiy filter issue have only entities with card-mod which causes flickering. Unfortunately, I do not know, how to present this behaviour in a shorter way.

@pickonedev: Do you perhaps have a more appropriate "minimal working example" to help @ildar170975 investigate this issue with the entity filter card?

**Comment by ildar170975** [comment]
Well, ofc I may try with a huge example - it is just a matter of time & a user's persistence... Providing a short MWE highly increases chances to localize the issue; also, there is a high probability that it will be YOU who localizes the issue while finding an MWE.
Ofc this is not just about the particular @ildar170975 - it is about all other users who may have a desire & spare time to investigate.

**Comment by RobusX** [comment]
Thank you very much! And yes, the entity-filter card is much more responsive when lights are blinking quickly compared to auto-entities card. Therefore, it would be great if this issue could be resolved.

**Comment by pickonedev** [comment]
About sections, was not so hard to do, at least not for me, and imagine that I have 7 similar dashboards (one per each room, with no layout differences, only the switches/sensors were different).
In the old dashboards, I used a horizontal three columns layout and converting to sections, I only had one issue of displaying something, I will show you above.
I just went to the original dashboard, click convert to sections and HA made a different dashboard converted to sections, then I just needed to modify some card_mod things, to make it identical like in the past. 

Look, this was the original dashboard

![image](https://github.com/user-attachments/assets/9c33a4ea-081d-4f44-bd1d-5f05c37d4dbd)

And after the conversion, I had only one issue, you can see in the image above, but I solve it easy and it was even better

![image](https://github.com/user-attachments/assets/1c08af83-653b-4385-a121-9252f7a9a01c)

**Comment by RobusX** [comment]
Thank you for interesting research. Probably this info will help @ildar170975 to have more detailed info about the annoying flicker issue.  We keep our fingers crossed that this issue will be investigated and resolved.
@ildar170975 if you need any additional info, just let us know. Thank you !

**Comment by pickonedev** [comment]
> Thanks @ pickonedev . Can you confirm too that in dashboard editing mode it is no flickering at all ? Thanks for the info.

Sorry, reading all the past comments, I now have understood exactly what you asked me here and I cannot confirm this.
It is happening in edit mode as well.

**Comment by RobusX** [comment]
OK, then we do not have similar issues in this case. 
In my situation, there is no flickering in other cards containing card mod when the dashboard edit mode is enabled (gray strip at the top, pencil pressed). 
However, as soon as I press the DONE button, flickering in other cards resumes, synchronized with the bulbs appearing and disappearing within the entity-filter card.

**Comment by ildar170975** [comment]
Is anyone going to provide an MWE ? )))

**Comment by pickonedev** [comment]
Can you please tell me what MWE is and how can I provide it?

**Comment by ildar170975** [comment]
@pickonedev 
Tested your case here:
```
  - title: 22x
    path: 22x
    type: custom:horizontal-layout
    cards:
      - type: entity-filter
        entities:
          - entity: switch.test_switch
            name: Room Light 1
            card_mod:
              style:
                hui-generic-entity-row$: |
                  .info, .value{font-size:6px}
          - entity: switch.test_switch_2
            name: Room Light 2
            card_mod:
              style:
                hui-generic-entity-row$: |
                  .info, .value{font-size:6px}
        state_filter:
          - 'on'
        show_empty: false
        card:
          type: entities
      - type: entities
        entities:
          - entity: switch.test_switch
          - entity: switch.test_switch_2
        title: static
```
![fil](https://github.com/user-attachments/assets/1ef92d2f-ec09-41d3-ad8d-349f5799d53e)

If you mean "font-size changes from normal to small" - this is called "transition".
It is observed not only with entity-filter.
Consider it is as "normal": sometimes styles are applied after some period.
In some cases it may be decreased by adding `transition: none` to card-mod code, sometimes not )).

**Comment by pickonedev** [comment]
Yes, yes, that's the "issue", that "flicker".

But I was talking about this in early comments, I know that there is no real issue, I think is the time gap between card elements loads and card_mod applying the css style.

I don't think it is about transition, I already made some tests, and by the way, there is no other card with the same "issue"...
Using conditional card it is working flawless.... which made me think that the "issue" can be solved if filter card will load all the entities from the start, but uses a different method to hides the one which does not meet the conditions.
I don't know exactly, I just think that javascript it is loading the entities only after the state meet the condition (not before), and when an element is newly loaded, only after loads, card_mod will apply css.
These said, a css method could solve this. Filter card could load all the entities, hiding the non-meet conditions with (display:none), then, card_mod style will be applied even for those which are not displayed, and finaly when filter card will change the display to initial, for the entities which meet the conditions, there will not be any "flicker"...

I have seen the same "issue" when I used a browser addon for changing the css for websites, that css will be applied only after the elements are loaded, because that addon it is not a server side css, it is applied from the client side... Maybe it is the same reason for which that "flicker" is happening with card_mod and filter card. Even if card_mod it is installed on the server, it can't be applied in the same time with the actual cards, so a time gap will appear...

**Comment by ildar170975** [comment]
>  Filter card could load all the entities, hiding the non-meet conditions with (display:none), then, card_mod style will be applied even for those which are not displayed, and finaly when filter card will change the display to initial, for the entities which meet the conditions, there will not be any "flicker"...

Check another case:
```
type: entity-filter
entities:
  - entity: switch.test_switch
    name: Room Light 1
  - entity: switch.test_switch_2
    name: Room Light 2
state_filter:
  - "on"
show_empty: false
card:
  type: entities
  card_mod:
    style:
      .card-content hui-toggle-entity-row $:
        hui-generic-entity-row $: |
          .info {font-size: 6px;}
```
Here card-mod is defined on a card's level - but a style for the ".info" element is defined inside a "card-mod" element on a row's level:
![image](https://github.com/user-attachments/assets/16b9768c-4042-43e2-a62b-af3a774c35bb)
and the style is applied after a row is created.

But - in some cases it may be "workarounded" by an external js.
Take same card - but with a specified class:
```
type: entity-filter
entities:
  - entity: switch.test_switch
    name: Room Light 1
  - entity: switch.test_switch_2
    name: Room Light 2
state_filter:
  - "on"
show_empty: false
card:
  type: entities
  card_mod:
    class: entity_filter
```
and this js:
```
customElements.whenDefined('hui-entities-card').then(() => {
  const EntitiesCard = customElements.get('hui-entities-card');
  const { html, css } = EntitiesCard.prototype;
  
  const newStyle = css`
  ha-card.entity_filter {font-size: 8px; transition: none;}`;
  
  const newStyles = [].concat(EntitiesCard.styles, newStyle);
  Object.defineProperty(EntitiesCard, 'styles',        {value: newStyles, configurable: true, enumerable: false});
  Object.defineProperty(EntitiesCard, 'elementStyles', {value: newStyles, configurable: true, enumerable: false});
});
```
And compare - left with card-mod (1st code in my post), right with a class:
![filt](https://github.com/user-attachments/assets/9919772d-78c8-4e8b-913f-a5145e2dfe2e)

**Comment by pickonedev** [comment]
> But - in some cases it may be "workarounded" by an external js.

With the extenal js it is working flowless... So... this means that the issue is not from entity filter card, this "flicker" could be solved from card mod? By changing many things insite... It would be great if this could be implemented in somethinng, in order to not using external js. Not everybody knows Java script :-D

I know, changing the size is extremely visible, same as hiding elements, but this is the way some of us want to change them.

I would try to implement the workaround with the external js, for all my "issues", till something will be changed in the future.

Thank you for your time.

**Comment by ildar170975** [comment]
> How can I add this css into that external js file?

That is what I meant by "in some cases it may be" )))
Never tried entering a row from a card in js, actually))

**Comment by pickonedev** [comment]
Ah, alright, I understand. Anyway, thanks.
I still hope that somehow, that "flicker" will be solved in the future... not a big deal, but very annoying when you have devices with low performance...

**Comment by RobusX** [comment]
@ildar170975

I am not a SW guru, so I do not even know where to put your Javascript text, but i assumed this text was only useful for @pickonedev's issue.
Now i am providing minimum of the minimum MWE for my issue, where a hidden toggle switch under the fold entity row blinks

Could you please kindly look into the issue, simulate this scenario at your side and suggest a solution which would prevent flickering issue ?

```yaml

1. Entity filter card with only one light which blinks due to automation at the bottom:

type: entity-filter
title: Lights/Sensors ON
show_header_toggle: false
entities:
  - entity: light.kurilbulb1
    name: Kur_1
    tap_action:
      action: toggle
state_filter:
  - "on"
  - open
  - detected
  - operator: ">"
    value: 4
show_empty: false
sort:
  method: domain
card:
  type: glance
  title: " "


2. Custom fold entity row card with only one entity where hidden toggle switch on the right flickers:

type: entities
entities:
  - type: custom:fold-entity-row
    head:
      entity: switch.shelly_main_door_12v_switch_0
      name: Main Door
      toggle: false
      state_color: true
      card_mod:
        style: |
          :host {
            --card-mod-icon-color: 
            {% if is_state(config.entity, 'on') %}
              orange;
            {% else %}
              grey;
            {% endif %};
          }
          .entities .state-card-toggle .state-card {
            display: none !important;
          }
          ha-entity-toggle {
            display: none !important;
          }
    entities:
      - type: custom:template-entity-row
        entity: switch.shelly_main_door_12v_switch_0
        name: Main Door 12V Switch
        tap_action:
          action: toggle
        toggle: true
        secondary_info: last-updated
        secondary: >
          Last: {{
          states.switch.shelly_main_door_12v_switch_0.last_updated.astimezone().strftime('%m-%d
          %H:%M:%S') }}
title: ""
state_color: true
show_header_toggle: false



3. Automation which blinks kuril1 bulb:

alias: Light Kuril Blinking RED when Armed TEST
description: Light Kuril Blinking RED when Armed TEST
triggers:
  - entity_id:
      - switch.alarm_plug
    to: "on"
    trigger: state
actions:
  - choose:
      - conditions: []
        sequence:
          - repeat:
              sequence:
                - action: light.turn_on
                  target:
                    entity_id:
                      - light.kurilbulb1
                  data:
                    color_name: red
                    brightness_pct: 2
                - delay: "00:00:00.1"
                - action: light.turn_off
                  target:
                    entity_id:
                      - light.kurilbulb1
                  data: {}
                - delay: "00:00:00.1"
                - action: light.turn_on
                  target:
                    entity_id:
                      - light.kurilbulb1
                  data:
                    color_name: white
                    brightness: 2
                - delay: "00:00:00.1"
                - action: light.turn_off
                  target:
                    entity_id:
                      - light.kurilbulb1
                  data: {}
                - delay: "00:00:06"
              until:
                - condition: state
                  entity_id: switch.alarm_plug
                  state: "off"
mode: single

**Comment by pickonedev** [comment]
> I do not even know where to put your Javascript text

You create a abc.js file inside www folder, then you add a javascript resource to the dashboard, with the path /local/abc.js, after that, you refresh the page and it should work, but it is the worst solution in testing because sometimes you need to open the dashboard into another browser with private tab, in order to see the changes... it is worst than css...

**Comment by ildar170975** [comment]
How to test with js file:
1. Create js file, add a code, save. 
2. Add a line into configuration.yaml to import this js. Save. 
3. Reboot HA.
4. Clean browser’s cache. 

How to test with changed js file:
1. Change js file, save. 
2. Clean browser’s cache. 
3. Reload a page.

**Comment by RobusX** [comment]
As I suspected, and as it now turns out, @pickonedev and I are facing different issues ( while I have no problems when editing the dashboard, @pickonedev still observes flickering) , because your js code has no effect on my issue.

I tried with addition of javascript code in the newly created entity_filter.js, file and put it into /config/www folder.
In the existing ui-lovelace.yaml, where I also have url links from other cards i added:

  - url: /local/entity_filter.js
    type: module

I modified my existing entity-filter card with alarm lamp and added this code below at the end (i also tried with "entities" instead of "glance", issue remained the same):
....
...
..
card:
  type: glance
  title: " "
  card_mod:
    class: entity_filter

After restarting Home Assistant, when the alarm blinking light is displayed in the entity filter card, the issue of the flickering toggle button in the main door card (as well as in several other cards with Card Mod applied) remains unchanged.
Flickering in the same manner, as it was before js file.

@ildar170975 Could you please confirm this with simulation and reproduction at your side too ? Thank you.

**Comment by ildar170975** [comment]
@RobusX 
I had a look at your code
```
type: entity-filter
title: Lights/Sensors ON
show_header_toggle: false
entities:
  - entity: light.kurilbulb1
    name: Kur_1
    tap_action:
      action: toggle
state_filter:
  - "on"
  - open
  - detected
  - operator: ">"
    value: 4
show_empty: false
sort:
  method: domain
card:
  type: glance
  title: " "
```
and a bit confused since it looks like a result of chatgpt.
Sorry - but this code has no sense:
1. The line "title: Lights/Sensors ON" line is meaningless - it does not belong to entity-filter.
2. "show_header_toggle" - same.
3. "sort" - same.
4. "state_filter" containing values impossible for a `light` entity.

Why not just replacing this with a simplest
```
type: entity-filter
entities:
  - entity: input_boolean.test_boolean
state_filter:
  - "on"
card:
  type: glance
```
for testing where you can manually toggle that `input_boolean` as frequently as you wish?

**Comment by ildar170975** [comment]
As for a code with `fold-entity-row`:
1. "toggle: false" is not supported for this row, you probably took it from a different custom card.
2. "state_color: true" for a "head" makes no sense since you are styling a color by card-mod.
3. This whole line ".entities .state-card-toggle .state-card" makes no sense - again, probably taken from another card.
4. "secondary_info: last-updated" makes no sense - it is not supported by template-entity-row.

So, before reporting a bug - suggest to always clean-up a code to make it in accordance to Docs.

**Comment by RobusX** [comment]
@ildar170975:

Thank you for your time and look into my case.
Yes, i mixed similar entity-filter card, where also sensors where in it and other power switches, where i want to display those, whose power >4...so this remained in cleaned light card...and I you are right, I am also using chat gpt:). Regarding your example, I am not using masonry, and i believe you, probably it is not flickering as video shows in your printed card. 

But, I am interesting in my situation with my existing cards /code, what is issue in my case.
So, when i changed and cleaned my cards regarding your advice, flickering remained.
Currently I tried only with this (all other code cleaned) + automation above for kuril1 light blinking which affects to the entity-filter. IN previous HA versions all worked ok.

Could you test exactly such case at your side, please ? This time i removed above suggested card-mod with js link from the entity-filter card.

```yaml
type: entity-filter
entities:
  - entity: light.kurilbulb1
    name: Kur_1
    tap_action:
      action: toggle
state_filter:
  - "on"
card:
  type: glance


In the card below hidden toggle switch flickers in the rhythm of kuril1 bulb in the automation above (this is just an example, in other cards also icon blinks, slider,..etc.

type: entities
entities:
  - type: custom:fold-entity-row
    head:
      entity: switch.shelly_main_door_12v_switch_0
      name: Main Door
      card_mod:
        style: |
          :host {
            --card-mod-icon-color: 
            {% if is_state(config.entity, 'on') %}
              orange;
            {% else %}
              grey;
            {% endif %};
          }
          .entities .state-card-toggle .state-card {
            display: none !important;
          }
          ha-entity-toggle {
            display: none !important;
          }
    entities:
      - type: custom:template-entity-row
        entity: switch.shelly_main_door_12v_switch_0
        name: Main Door 12V Switch

**Comment by ildar170975** [comment]
```
  - title: ...
    path: ...
    cards:
      - type: entity-filter
        entities:
          - entity: light.xxx
            name: xxx
            tap_action:
              action: toggle
        state_filter:
          - 'on'
        card:
          type: glance
      - type: entities
        entities:
          - type: custom:fold-entity-row
            head:
              entity: switch.test_switch
              name: Main Door
              card_mod:
                style: |
                  :host {
                    --card-mod-icon-color: 
                    {% if is_state(config.entity, 'on') %}
                      orange;
                    {% else %}
                      grey;
                    {% endif %};
                  }
                  ha-entity-toggle {
                    display: none !important;
                  }
            entities:
              - type: custom:template-entity-row
                entity: switch.test_switch
                name: Main Door 12V Switch
    type: masonry
```
![xcv](https://github.com/user-attachments/assets/408a960c-dbbf-47e0-8fe4-b889b9ec76e0)
Well, you are toggling a `light` entity every 100ms -> the `entity-filter` card is redrawn -> a card below is redrawn as well -> and you see a transitional effects as expected since 100ms appears to be a small interval on your client.

**Comment by RobusX** [comment]
Thank you very much for your research and interesting findings.
So, if i assume correctly, solution would be only to increase blinking period within automation, or using sections.
Regarding first, I just tried with 200ms, 500ms delay, and 1s delay and flickering continued all the way, but i cannot increase it near the 1second range, because then feel of "alarmed place" when looking into blinking lamps is fading out...probably i should say "goodbye" to blinking lamps in the dashboard. 
Entity filter card was really responsive in the past, compared to auto-entities.

Regarding sections. I am not familiar with sections yet, so i am in doubt how will this look in the mobile phone, where only one row is displayed at once...and also limitation with equal size quadrants.
if it is possible to include all entities (multiple, custom, and entity filter) into the sections grid and will work without flickering...but even then, equal size limitation probably.
What about masonry placement like here at the link below which allows cards with different heights within one row?
https://www.home-assistant.io/dashboards/masonry/
Will this cause flickering too ?
Thanks again !

**Comment by ildar170975** [comment]
> Entity filter card was really responsive in the past, compared to auto-entities.

Any card with a changing width seems to cause a redraw -> flickering.

> using sections.

Could be the way. Not using sections myself; seems that they have a kind of fixed placement which in your case prevents a redraw of sibling cards.

> Regarding sections

Erm, can say you only a little about sections. And imho this is beyond this card-mod issue (which I think should be closed).

> What about masonry placement 

My 1st flickering example above was with masonry. Ofc if a topmost card (entity-filter) changes it's height -> all other cards will be redrawn.

**Comment by pickonedev** [comment]
3 months later, still no solution? :-(
